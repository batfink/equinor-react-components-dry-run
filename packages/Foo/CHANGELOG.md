# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.2.0"></a>
# [0.2.0](https://github.com/batfink/equinor-react-components-dry-run/compare/@equinor-internal/foo@0.0.2...@equinor-internal/foo@0.2.0) (2018-09-12)


### Features

* change color ([588a2b2](https://github.com/batfink/equinor-react-components-dry-run/commit/588a2b2))
* update colors ([b456d3d](https://github.com/batfink/equinor-react-components-dry-run/commit/b456d3d))





<a name="0.1.0"></a>
# [0.1.0](https://github.com/batfink/equinor-react-components-dry-run/compare/@equinor-internal/foo@0.0.2...@equinor-internal/foo@0.1.0) (2018-09-11)


### Features

* change color ([588a2b2](https://github.com/batfink/equinor-react-components-dry-run/commit/588a2b2))
