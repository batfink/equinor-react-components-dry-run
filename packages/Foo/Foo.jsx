import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Bar from '@equinor-internal/bar'

const Foo = ({ className, ...props }) =>
    (<Wrapper className={ className } { ...props }><Bar /></Wrapper>)

const Wrapper = styled.div`
    padding: 20px;
    background: violet;
    position: relative;
    font: 12px sans-serif;

    &::before {
        position: absolute;
        content: 'Foo';
        top: 2px;
        left: 2px;
        padding: 0.2em;
        background: rgba(255,255,255,0.8);
    }
`

Foo.propTypes = {
    className: PropTypes.string,
}

Foo.defaultProps = {
    className: '',
}

export default Foo
