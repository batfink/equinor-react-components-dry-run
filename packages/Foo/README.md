Foo
====

Usage
-----

```bash
yarn add @equinor-internal/foo
```
