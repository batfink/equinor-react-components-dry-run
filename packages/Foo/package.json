{
    "name": "@equinor-internal/foo",
    "main": "Foo",
    "version": "0.3.1",
    "author": "Victor Nystad",
    "license": "UNLICENSED",
    "dependencies": {
        "@equinor-internal/bar": "^0.2.1",
        "prop-types": "^15.6.2",
        "react": "^16.5.0",
        "styled-components": "^3.4.5"
    }
}
