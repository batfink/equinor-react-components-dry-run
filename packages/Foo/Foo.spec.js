import React from 'react'
import Foo from './Foo'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('Foo', () => {
    it('should render same as snapshot', () => {
        const tree = renderer.create(<Foo />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
