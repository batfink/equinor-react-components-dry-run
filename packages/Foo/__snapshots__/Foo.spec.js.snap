// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo should render same as snapshot 1`] = `
.c1 {
  padding: 20px;
  background: green;
  position: relative;
  font: 12px sans-serif;
}

.c1::before {
  position: absolute;
  content: 'Bar';
  top: 2px;
  left: 2px;
  padding: 0.2em;
  background: rgba(255,255,255,0.8);
}

.c0 {
  padding: 20px;
  background: red;
  position: relative;
  font: 12px sans-serif;
}

.c0::before {
  position: absolute;
  content: 'Foo';
  top: 2px;
  left: 2px;
  padding: 0.2em;
  background: rgba(255,255,255,0.8);
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
